script/files/boot/dts/overlays/poweroff-pins.dts

```diff
@@ -0,0 +1,26 @@
+/dts-v1/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
+	fragment@0 {
+		target-path = "/";
+		__overlay__ {
+			poweroff_gpio: poweroff_gpio {
+				pinctrl-names = "default";
+				pinctrl-0 = <&poweroff_pins>;
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			poweroff_pins: poweroff_pins {
+				brcm,pins = <23 24>;
```

Comment:
> the pin # is 23 and 24, but the GPIO label are 16 and 18

I think you might have that part reversed. According to the pinout (e.g. [pinout.xyz](https://pinout.xyz)), the physical pin numbers are 16 and 18 and the "BCM" numbers are [23](https://pinout.xyz/pinout/pin16_gpio23) and [24](https://pinout.xyz/pinout/pin18_gpio24).

That said, where the kernel docs for the GPIO controller are from Broadcom, I think they're using the BCM numbers. Admittedly, a lot of what's done here is based off of [the lirc-rpi dtoverlay in the Raspberry Pi folks' kernel tree](https://github.com/raspberrypi/linux/blob/rpi-4.4.y/arch/arm/boot/dts/overlays/lirc-rpi-overlay.dts) (you can see references to it in the [Device Trees, overlays, and parameters](https://www.raspberrypi.org/documentation/configuration/device-tree.md) document).

See also: [1](http://raspberrypi.stackexchange.com/a/12967), [2](https://www.kernel.org/doc/Documentation/devicetree/bindings/pinctrl/brcm,bcm2835-gpio.txt)